daffodil-udf/src/main/java/org/apache/daffodil/udf/FunctionClassInfo.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.udf;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface FunctionClassInfo {
```

Comment:
I am a bit mystified why we need this annotation. 
Static methods of the class won't do? Inheriting from an interface DaffodilUDFDefinition? 

If so please add javadoc here as to why this annotation is needed. Are we using it to identify the classes on the classpath that are supposed to be UDFs? If so then it doesn't need to have these hard-coded hard-to-maintain strings in it. It could just be an annotation that's a tag with no arguments. I suggest that because I really would rather we somehow computed the namespace and name from the class itself.

Reply:
If we do keep this, it might make sense to rename it UserDefineFunctionInfo to be consisent with UserDefinedFunctionProvider